### PR TITLE
Backport/unsupported queries

### PR DESCRIPTION
**Pull request by electrachong**
Depends on https://github.com/scality/Integration/pull/539

Backport from master (rel/7.0):
- fix: checks for unsupported queries in request and returns NotImplemented if any are found
- upgrades s3cmd to v1.6.1 to match version expecting users to use
- skip s3cmd info tests failing because bucket location is not implemented (unskipped in rel/7.0)
- commit that includes related optimizations

### CODE REVIEW

**Comment by electrachong** [comment]
@ironman-machine

**Comment by electrachong** [comment]
@ironman-machine

**Comment by electrachong** [comment]
end-to-end retried here with integration branch: http://ci.ironmann.io/gh/scality/Integration/13494

**Comment by electrachong** [comment]
I remembered that even if we upgrade the s3cmd version, the s3cmd info still fail because bucket location is not implemented. (For some reason, s3cmd returns error for entire command if bucket location is not implemented, but is fine if any of the other requests get a NotImplemented response. This was a problem we ran into in master branch, but we managed to merge the bucket location  feature before this commit.) Skipping the related s3cmd tests.

@ironman-machine
SCALITY_INTEGRATION_BRANCH=backport/upgrade-s3cmd

**Comment by electrachong** [comment]
end-to-end passed

File: lib/routes/routesUtils.js [hunk 1]
```diff
@@ -100,11 +100,19 @@ function errorXMLResponse(errCode, response, log, corsHeaders) {
...
+        // eslint-disable-next-line no-param-reassign
+        corsHeaders['Content-Length'] = xmlStr.length;
+    }
     setCommonResponseHeaders(corsHeaders, response, log);
     response.writeHead(errCode.code, { 'Content-type': 'application/xml' });
     return response.end(xmlStr, 'utf8', () => {
         log.end().info('responded with error XML', {
             httpCode: response.statusCode,
+            xmlStr,
+            corsHeaders,
```
Comment (LaurenSpiegel) [inline lib/routes/routesUtils.js]: please remove these two lines.  I did a subsequent pr on master to remove these.

Comment (electrachong) [inline lib/routes/routesUtils.js]: I updated to remove the lines from the logs

**Pull request merged**